package-res/resources/web/pentaho/data/Filter.js

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./filter/IsEqual",
+  "./filter/IsIn",
+  "./filter/And",
+  "./filter/Or",
+  "./filter/Not"
+], function(IsEqual, IsIn, AndFilter, OrFilter, NotFilter) {
+  "use strict";
+
+  /**
+   * @name RootFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @ignore
+   *
+   * @classdesc The `RootFilter` class implements a type of
+   * `AbstractTreeFilter` {@link pentaho.data.filter.AbstractTreeFilter}. Specifically
+   * the `RootFilter` extends the `RootFilter` {@link pentaho.data.filter.Or} filter.
+   *
```

Comment:
Use the tag @extends to provide the information on hierarchy.
I think the @classdesc tag should explain the what and the why and not so much the how.

The RootFilter class is not really exposed, and is mostly an artifice to create a filter tree from a spec.
Perhaps it should simply drop dead.